keel-api/src/main/kotlin/com/netflix/spinnaker/keel/api/plugins/ResourceHandler.kt

```diff
@@ -137,6 +153,26 @@ interface ResourceHandler<S : ResourceSpec, R : Any> : SpinnakerExtensionPoint {
    */
   @JvmDefault
   suspend fun actuationInProgress(resource: Resource<S>): Boolean = false
+
+  /**
+   * Notifies core Keel that a new artifact version is being deployed to the specified resource.
+   *
+   * The default implementation achieves this by publishing an [ArtifactVersionDeploying] event via the
+   * [EventPublisher], and should *not* be overridden by plugin implementations.
+   */
+  @JvmDefault
+  fun notifyArtifactDeploying(resource: Resource<S>, artifactVersion: String) =
```

Comment:
I recommend marking these new methods as `protected` unless you foresee a scenario where they'll be called by anyone other than subclasses.

Reply:
Can't because it's an interface...